### PR TITLE
README.md for SSH

### PR DESCRIPTION
**Pull request by cydia2020**

### CODE REVIEW

File: tools/ssh/README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+![SSH into comma two](https://user-images.githubusercontent.com/37757984/82586797-0496cc00-9b4d-11ea-9e98-48d193cf38ff.jpg)
```
Comment (adeebshihadeh) [inline tools/ssh/README.md:1]: remove the "comma two" references, think this should be more generic

Comment (adeebshihadeh) [reply tools/ssh/README.md:1]: also don't think we need the image

File: tools/ssh/README.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+# Connecting To Your comma two Using SSH
```
Comment (adeebshihadeh) [inline tools/ssh/README.md:2]:
```suggestion
# Connecting to your comma device over SSH
```

File: tools/ssh/README.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+This will walk you through the procedure of connecting to your comma two using SSH in order for you to make localised changes on your comma two.
+
+Table of Contents
```
Comment (adeebshihadeh) [inline tools/ssh/README.md:5]: remove table of contents, this should be short

File: tools/ssh/README.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+=================
+
+* [Creating a GitHub Account](#create-github-account)
+* [Generating an SSH Key](#generate-ssh-key)
+* [Uploading the Public SSH Key To Your GitHub Account](#upload-public-key)
+* [Register Your Public Key On Your comma two](#register-key-on-c2)
+* [Connecting To Your comma two Using SSH](#connection)
+
+# Creating a GitHub Account<a name="create-github-account" />
```
Comment (adeebshihadeh) [inline tools/ssh/README.md:14]: we don't need steps for this

File: tools/ssh/README.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+In order to create a GitHub Account, you have to have your email address handy and ready to receive your confirmation email.
+1. Go to the [GitHub Account Creation Page](https://github.com/join).
+2. Enter your information, this includes your desired username, email address, and password.
+3. Once your details are correct, you may click the `create account` button.
+4. You will soon receive an account verification email from GitHub, once your account is verified, you may proceed to the next step.
+
+For more information on email verification, please see the [GitHub Doc](https://docs.github.com/en/github/getting-started-with-github/verifying-your-email-address) regarding this topic.
+
+# Generating an SSH Key<a name="generate-ssh-key" />
```
Comment (adeebshihadeh) [inline tools/ssh/README.md:23]: github already has a comprehensive guide for this, just link to it

File: tools/ssh/README.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+An SSH key pair is like a padlock and its key, your public key is the padlock, and your private key holds the information of opening the padlock.
+
+To create your SSH key pair, you will have to have OpenSSH client installed on your operating system, please consult your operating system manual on the procedure of accomplishing this.
+
+Once OpenSSH Client is installed and operational, using command line console, you will have to type `ssh-keygen -t rsa -f ~/.ssh/id_rsa`, it will then prompt you for a password, you may ignore this and press the `enter` key; nevertheless, setting a password is always a good practice.
+
+This will create two files in your `.ssh` folder, one file will be named `id_rsa` and another file will be named `id_rsa.pub`. This step is completed once these two files are present in your `.ssh` folder.
+
+# Uploading the Public SSH Key To Your GitHub Account<a name="upload-public-key" />
```
Comment (adeebshihadeh) [inline tools/ssh/README.md:32]: again, github guide should be sufficient here

**Comment by adeebshihadeh** [comment]
Let's add a link to this README in tools/README.md

**Comment by pd0wm** [comment]
Maybe we should also make explicit that when you set your github username it only fetches your keys *once*? So any keys you add/change on github after that won't be reflected on the device.

File: tools/README.md [hunk 7]
```diff
@@ -65,7 +65,10 @@ Review video and log data from routes and stream CAN messages to your device.
 
 Use a joystick to control your car.
 
+[SSH](ssh)
```
Comment (adeebshihadeh) [inline tools/README.md:68]: let's move this above the tools section, it's not really a tool

File: tools/ssh/README.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
+# Connecting to your comma device using SSH
+This will walk you through the procedure of connecting to your comma device using SSH in order for you to make localised changes on your comma device.
+
+Your comma device fetches your public SSH keys from your GitHub account. You will have to have a pre-existing GitHub Account, for more information on account registration, visit [this GitHub Doc](https://docs.github.com/en/github/getting-started-with-github/signing-up-for-a-new-github-account).
+
+Your email will have to be verified by GitHub, for more information on email verification, please see the [GitHub Doc](https://docs.github.com/en/github/getting-started-with-github/verifying-your-email-address) regarding this topic.
```
Comment (adeebshihadeh) [inline tools/ssh/README.md:6]: remove this

File: tools/ssh/README.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
+# Connecting to your comma device using SSH
+This will walk you through the procedure of connecting to your comma device using SSH in order for you to make localised changes on your comma device.
```
Comment (adeebshihadeh) [inline tools/ssh/README.md:2]: remove this

File: tools/ssh/README.md [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+# General information
+For security reasons, SSH is not enabled by default on your comma device, you can enable it by navigating to `Settings → Developer → Enable SSH`.
+
+SSH Keys can be imported from GitHub by going to `Settings → Developer → GitHub Username`, once GitHub username is inserted, it will automatically fetch the public keys on your account.
+
+default ssh username: `root`
+
+default ssh port: `8022`
+
+To view the IP address of your comma device, go to `Settings → WiFi → Open WiFi Settings → More Options → Three Dot Menu at the Top Left Corner → Advanced`.
```
Comment (adeebshihadeh) [inline tools/ssh/README.md:17]:
```suggestion
# Quick Start

SSH is disabled by default. SSH can be enabled in the Network section of the settings. The IP address can also be found in the settings.

* Username: root
* Port: 22 or 8022
```

**Comment by ErichMoraga** [comment]
"Three Dots Menu at the Top Left Corner"

That should be the top-right (which technically doesn't require capitalization).

**Comment by cydia2020** [comment]
> "Three Dots Menu at the Top Left Corner"
> 
> That should be the top-right (which technically doesn't require capitalization).

Oooh yes, my brain got stuck when writing this, will fix in a bit

**Pull request merged**